test/WebJobs.Extensions.Tests/Extensions/ApiHub/ApiHubBindingEndToEndTests.cs

```diff
@@ -0,0 +1,306 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.ApiHub;
+using Microsoft.Azure.WebJobs.Extensions.ApiHub;
+using Microsoft.Azure.WebJobs.Extensions.Tests.Common;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Tests.Extensions.ApiHub
+{
+    public class ApiHubBindingEndToEndTests
+    {
+        private const string ImportTestPath = @"import";
+        private const string OutputTestPath = @"output";
+
+        private string _apiHubConnectionString;
+        private IFolderItem _rootFolder;
+
+        public ApiHubBindingEndToEndTests()
+        {
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AzureWebJobsDropBox")))
+            {
+                _apiHubConnectionString = Environment.GetEnvironmentVariable("AzureWebJobsDropBox");
+            }
+            else
+            {
+                _apiHubConnectionString = "UseLocalFileSystem=true;Path=" + Path.GetTempPath() + "ApiHubDropBox";
+            }
+
+            _rootFolder = ItemFactory.Parse(_apiHubConnectionString);
+
+            var folder = _rootFolder.CreateFolderAsync(ImportTestPath).GetAwaiter().GetResult();
+            foreach (var item in folder.ListAsync(true).GetAwaiter().GetResult())
+            {
+                var i = item as IFileItem;
+                if (i != null)
+                {
+                    i.DeleteAsync();
+                }
+            }
+
+            folder = _rootFolder.CreateFolderAsync(OutputTestPath).GetAwaiter().GetResult();
+            foreach (var item in folder.ListAsync(true).GetAwaiter().GetResult())
+            {
+                var i = item as IFileItem;
+                if (i != null)
+                {
+                    i.DeleteAsync();
+                }
+            }
+
+            ApiHubTestJobs.Processed.Clear();
+        }
+
+        [Fact]
+        public async void JobIsTriggeredForNewFiles()
+        {
+            JobHost host = CreateTestJobHost();
+
+            host.Start();
+
+            Assert.Equal(0, ApiHubTestJobs.Processed.Count);
+
+            // now write a file to trigger the job
+            var fileItem = await WriteTestFile();
+
+            // waiting for 20 sec to make sure ApiHub trigger is fired.
+            await Task.Delay(20000);
```

Comment:
Also, is there a way to have this test not take 20 seconds? Since it's a unit test, I'd expect that it should run very quickly. Perhaps if you make your 15 second poll interval configurable (maybe just for tests for now), you can set the poll interval to 2 seconds to make the test fast.